website/docs/guides/semantic-nullability.md

```diff
@@ -0,0 +1,107 @@
+---
+id: semantic-nullability
+title: "Semantic Nullability"
+slug: /guides/semantic-nullability/
+description: Experimental support for semantic nullability
+keywords:
+- "null"
+- nullability
+- semanticNonNull
+---
+
+:::warning
+**Experimental**: Because Strict Semantic Nullability is still in flux, the implementation and behavior within Relay are subject to change and may have unexpected behavior as we learn more about the idea and its implications. 
+:::
+
+## Motivation
+
+One of GraphQL's strengths is its field-granular error handling which can dramatically improve response resiliency. However, today that error handling depends upon field nullability, which is the reason it is a [recommended best practice](https://graphql.org/learn/best-practices/#nullability) to default all fields to being nullable. This creates a trade-off where __enabling maximum resiliency means client developers must manually handle all possible permutations of field nullability__ within their components. [`@required`](./required-directive.md) can help a bit, but is ultimately a very blunt tool.
+
+## Proposed Solution
+
+[Semantic Nullability](https://github.com/graphql/graphql-wg/discussions/1410) is an early GraphQL spec proposal that aims to decouple error handling and nullability in the GraphQL spec to enable maximum resiliency while still exposing the "semantic nullability", (the nullability of the actual resolver function/method on the server) of the field to the client.
```

Comment:
@benjie I wonder what the canonical link should be here. Lee's post is not the most up to date. Should this be your discussion post https://github.com/graphql/nullability-wg/discussions/58 maybe?

Reply:
This is probably the best link right now:

https://github.com/graphql/graphql.github.io/blob/nullability-post/src/pages/blog/2024-08-14-exploring-true-nullability.mdx#our-latest-proposal

I need to either update what's being proposed in the issue you linked, or alternatively propose a slightly different alternative (more like what's in this blog post above).

Essentially what I'm proposing now is that users (people writing the web frontends/mobile apps/etc) would only ever consume a traditional-looking schema with traditional behaviors, but that schema may be different to the one that the server exposes because their client of choice (Relay, Apollo, URQL, etc) has factored in both how it handles nullability (e.g. "throw on read" would allow treating the semantically non-nullable types as if they were strictly non-nullable) and any other client-specific details such as local schema extensions.

Reply:
To be clear: it's still essentially the same solution technically, it's just being presented differently - e.g. more carefully defining the audiences for each mode.